### PR TITLE
[GEOT-6647] Clean up and improve Jackson based gt-geojsondatastore

### PR DESCRIPTION
**Pull request by ianturton**
<Include a few sentences describing the overall goals for this Pull Request>

# Checklist

Reviewing is a process done by project maintainers, **mostly on a volunteer basis** (thus limited in time). We need to keep the review overhead as small as possible, and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md)
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not causing two modules to share the same Java packages (to avoid [Java 9+ split package](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed) issues)
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):

- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.

### CODE REVIEW

File: modules/unsupported/geojsonstore/.checkstyle [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aaime) [inline modules/unsupported/geojsonstore/.checkstyle:1]: Confused, why is this needed?

File: modules/unsupported/geojsonstore/src/main/java/com/bedatadriven/jackson/datatype/jts/GeoJson.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+/*
```
Comment (aaime) [inline modules/unsupported/geojsonstore/src/main/java/com/bedatadriven/jackson/datatype/jts/GeoJson.java:17]: Hum... is it sufficient to link the original source code or do we have to have both license headers?
But I see, the original code did not have license headers either: 
https://github.com/bedatadriven/jackson-datatype-jts/blob/master/src/main/java/com/bedatadriven/jackson/datatype/jts/GeoJson.java

Thinking out loud here: do we need to preserve the original package names?

File: modules/unsupported/geojsonstore/.checkstyle (hunk 1 above)
Comment (ianturton) [reply modules/unsupported/geojsonstore/.checkstyle:1]: Possibly something Intellij did for me

File: modules/unsupported/geojsonstore/src/main/java/org/geotools/data/geojson/GeoJSONReader.java [hunk 3]
```diff
@@ -128,6 +129,44 @@ public static SimpleFeature parseFeature(String json) throws JsonParseException,
...
+     */
+    public static SimpleFeatureCollection parseFeatureCollection(String jsonString) {
+
+        try (GeoJSONReader reader =
+                new GeoJSONReader(new ByteArrayInputStream(jsonString.getBytes()))) {
+            SimpleFeatureCollection features = (SimpleFeatureCollection) reader.getFeatures();
+            return features;
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
```
Comment (aaime) [inline modules/unsupported/geojsonstore/src/main/java/org/geotools/data/geojson/GeoJSONReader.java:144]: Nope? :-D

I would have thought we'd have a QA rule against this, but nope, it's not there:
https://github.com/geotools/geotools/blob/master/build/qa/pmd-ruleset.xml
It could, however, be added. Will try to remember about it (not asking you to do it and fix all issues, you have enough on your plate with this geojson switch):
https://pmd.github.io/pmd-6.31.0/pmd_rules_java_bestpractices.html#avoidprintstacktrace

File: modules/unsupported/geojsonstore/src/main/java/com/bedatadriven/jackson/datatype/jts/GeoJson.java (hunk 2 above)
Comment (ianturton) [reply modules/unsupported/geojsonstore/src/main/java/com/bedatadriven/jackson/datatype/jts/GeoJson.java:17]: @jodygarnett recommended this in the previous PR. As for the package names I didn't think of changing them but there is no reason not to I guess.

**Comment by jodygarnett** [comment]
Reviewing PRs ahead of release candidate, any chance this draft can make it in?

**Comment by ianturton** [comment]
OK, I'm ready to push this one for 25-RC, the existing GeoJSON is not deprecated as I don't have time to go through the GeoServer code base and update everything, but this is ready to go in GeoTools
